src/derivative_operators/abstract_operator_functions.jl

```diff
@@ -164,7 +166,12 @@ end
 ################################################################################
 
 function *(coeff_func::Function, A::DerivativeOperator{T,N,Wind}) where {T,N,Wind}
-    coefficients = A.coefficients === nothing ? Vector{T}(undef,A.len) : A.coefficients
+    if hasmethod(coeff_func, (Float64,)) || hasmethod(coeff_func, (Vector{Float64},))
```

Comment:
Why Float64? That's not gonna fly. Also, `hasmethod` will be really fragile, do not rely on it.